character-set-and-collation.md

```diff
@@ -6,7 +6,15 @@ aliases: ['/docs/dev/character-set-and-collation/','/docs/dev/reference/sql/char
 
 # Character Set and Collation
 
-A character set is a set of symbols and encodings. A collation is a set of rules for comparing characters in a character set.
+This document introduces the character set and collation supported by TiDB.
+
+## Concepts
+
+A character set is a set of symbols and encodings. 
+
+A collation is a set of rules for comparing characters in a character set.
```

Comment:
May I suggest the following as an intro:

A character set is a set of symbols and encodings. The default character set in TiDB is utf8mb4, which matches the default in MySQL 8.0 and above. In TiDB `utf8` and `utf8mb4` behave identically, and `utf8` is not restricted to a maximum of 3 bytes as in MySQL. UTF-8 encoding [accounts for between 83% - 100% of webpages](https://en.wikipedia.org/wiki/UTF-8), depending on the language and country.

A collation is a set of rules for comparing characters in a character set, and the sorting order of characters. For example in a binary collation `A` and `a` do not compare as equal:

{{< copyable "sql" >}}

```sql
SET NAMES utf8mb4 COLLATE utf8mb4_bin;
SELECT 'A' = 'a';
SET NAMES utf8mb4 COLLATE utf8mb4_general_ci;
SELECT 'A' = 'a';
```

```sql
mysql> SELECT 'A' = 'a';
+-----------+
| 'A' = 'a' |
+-----------+
|         0 |
+-----------+
1 row in set (0.00 sec)

mysql> SET NAMES utf8mb4 COLLATE utf8mb4_general_ci;
Query OK, 0 rows affected (0.00 sec)

mysql> SELECT 'A' = 'a';
+-----------+
| 'A' = 'a' |
+-----------+
|         1 |
+-----------+
1 row in set (0.00 sec)
```

TiDB defaults to using a binary collation. This differs from MySQL, which uses a case-insensitive collation by default.

Reply:
@wjhuang2016 PTAL

Reply:
In TiDB utf8 and utf8mb4 behave identically, and utf8 is not restricted to a maximum of 3 bytes as in MySQL.

In fact, TiDB checks if the bytes character is greater than 3 unless `check-mb4-value-in-utf8` is disabled.

Reply:
@nullnotnil Do you need to modify your suggested intro according to https://github.com/pingcap/docs/pull/3402#discussion_r459830078?

Reply:
I've updated it.

Reply:
> @nullnotnil Do you need to modify your suggested intro according to [#3402 (comment)](https://github.com/pingcap/docs/pull/3402#discussion_r459830078)?

Yes, I verified and my earlier comment was incorrect. I have updated it to describe the check-mb4 option. I will add a section on utf8mb4 vs utf8.